lua/roslyn/init.lua

```diff
@@ -210,44 +211,51 @@ function M.setup(config)
+
                 return
             end
 
-            -- We need to always check if there is more than one solution. If we have this check below the check for the pipe name
-            -- Then we wouldn't give the users an option to change target if they navigate to a different project with multiple solutions
-            -- after they have already started the roslyn language server, and a solution is chosen
-            local targets = vim.fn.glob(vim.fs.joinpath(sln_dir, "*.sln"), true, true)
-            if #targets > 1 then
-                vim.notify_once(
-                    "Multiple targets found. Use `CSTarget` to select target for buffer",
-                    vim.log.levels.INFO
-                )
+            local all_sln_files = require("roslyn.slnutils").get_all_solution_files(opt.buf)
 
-                vim.api.nvim_create_user_command("CSTarget", function()
-                    vim.ui.select(targets, { prompt = "Select target: " }, function(target)
-                        solution[sln_dir] = target
-                        if _pipe_name then
-                            lsp_start(_pipe_name, target, roslyn_config)
-                        else
-                            run_roslyn(exe, target, roslyn_config)
-                        end
-                    end)
-                end, { desc = "Selects the target for the current buffer" })
+            if not all_sln_files then
+                return
             end
 
-            -- Roslyn is already running, so just call `vim.lsp.start` to handle everything
-            if _pipe_name and solution[sln_dir] then
-                return lsp_start(_pipe_name, solution[sln_dir], roslyn_config)
+            vim.api.nvim_create_user_command("CSTarget", function()
+                vim.ui.select(all_sln_files, { prompt = "Select target solution: " }, function(sln_file)
+                    known_solutions[sln_directory] = sln_file
+                    if _pipe_name then
+                        lsp_start(_pipe_name, sln_file, roslyn_config)
+                    else
+                        run_roslyn(exe, sln_file, roslyn_config)
+                    end
+                end)
+            end, { desc = "Selects the sln file for the current buffer" })
+
+            if #all_sln_files == 1 then
+                run_roslyn(exe, all_sln_files[1], roslyn_config)
+                known_solutions[sln_directory] = all_sln_files[1]
+
+                return
             end
 
-            if #targets == 1 then
-                run_roslyn(exe, targets[1], roslyn_config)
-                solution[sln_dir] = targets[1]
+            -- Multiple SLN files found, let's try to predict which one is the correct one for the current buffer
+            local predicted_sln_file = require("roslyn.slnutils").get_current_solution_file(opt.buf)
+
+            if predicted_sln_file then
+                run_roslyn(exe, predicted_sln_file, roslyn_config)
```

Comment:
Should we do a `vim.notify` here about which solution file was predicted? I don't really have any strong preferences for this as I believe a couple of users could probably see this as noice, but I can also see that it could be valuable to know. What do you think?

Reply:
Right, initially I also thought I should add such a notification, but then I realized that there already is such a thing:
![image](https://github.com/seblj/roslyn.nvim/assets/10273406/43fa621e-3d4a-4dc7-9fbc-c24de48d547c)

This notification always pops up, so I guess it's sufficient.